### PR TITLE
ipa_user: Add userauthtype param

### PR DESCRIPTION
**Pull request by linahe**
##### SUMMARY

Add `userauthtype` param to allow setting an IPA user's authentication type.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ipa_user
##### ADDITIONAL INFORMATION

[Freeipa client API docs for ipauserauthtype param](https://python-freeipa.readthedocs.io/en/latest/index.html?highlight=user_mod#python_freeipa.client_meta.ClientMeta.user_mod)

```paste below

```

### CODE REVIEW

File: changelogs/fragments/951-ipa_user-add-userauthtype-param.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - "ipa_user - add ``userauthtype`` param (https://github.com/ansible-collections/community.general/pull/951)."
```
Comment (Andersson007) [inline changelogs/fragments/951-ipa_user-add-userauthtype-param.yaml:3]:
```suggestion
  - "ipa_user - add ``userauthtype`` option (https://github.com/ansible-collections/community.general/pull/951)."
```
according to the doc `option` is more preferable than `parameter` or `argument`

File: plugins/modules/identity/ipa/ipa_user.py [hunk 2]
```diff
@@ -90,6 +90,11 @@
     - Default home directory of the user.
     type: str
     version_added: '0.2.0'
+  userauthtype:
+    description:
+    - The authentication type to use for the user.
+    choices: ["password", "radius", "otp", "pkinit", "hardened"]
+    type: str
```
Comment (Andersson007) [inline plugins/modules/identity/ipa/ipa_user.py:97]:
```suggestion
    type: str
    version_added: '1.2.0'
```

**Review by Andersson007 — COMMENTED** [review]
@linahe thanks for the PR
@Akasurde could you please take a look?

**Comment by Andersson007** [comment]
@Nosmoht , as an original author, could you please look at the PR

**Comment by Andersson007** [comment]
@linahe please add an example with new option to the `EXAMPLE` block

**Review by Andersson007 — APPROVED** [review]
LGTM
imo, after the example added, we could merge this. What do you think @felixfontein ?

**Comment by Andersson007** [comment]
(@linahe the release is going to be tomorrow, so would be cool to have the example here today)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@linahe thanks for implementing this!

**Comment by Andersson007** [comment]
@linahe yep, thanks for the feature!
BTW, there are 2 other feature ideas related to `ipa_`* modules https://github.com/ansible-collections/community.general/labels/identity , in case you're interested in continuing the contribution to Ansible;)

**Pull request merged**